deployer/gateway_fqdn_deployer.go

```
@@ -79,11 +79,9 @@ func (k *GatewayFQDNDeployer) Deploy(ctx context.Context, gw *workloads.GatewayF
 	currentDeployments, err := k.deployer.Deploy(ctx, oldDeployments, newDeployments, newDeploymentsData, newDeploymentsSolutionProvider)
 
 	// update state
-	if err == nil {
-		gw.ContractID = currentDeployments[gw.NodeID]
-		gw.NodeDeploymentID = currentDeployments
-		k.tfPluginClient.StateLoader.currentNodeDeployment[gw.NodeID] = gw.ContractID
-	}
+	gw.ContractID = currentDeployments[gw.NodeID]
```

why removing the error check? also we can return the error if exists